particle-iot/device-os
# feature/sc-101084/e404x-platform-toolchain-manifest

### Problem

Local compiler toolchain does not recognize the new `esomx` platform

### Solution

Add the new `esomx` platform id to the list of supported platforms for the `deviceOS@source` toolchain

### Steps to Test

1. Follow the [installation instructions](https://github.com/particle-iot/cli/tree/main/packages/cli#installation) for CLI-vNext (aka "Delorean")
2. Install the `device@source` toolchain: `prtcl toolchain:install source:/path/to/device-os-repo` (consult `--help` if you need)
3. View and validate the `device@source` toolchain: `prtcl toolchain:view source:/path/to/device-os-repo --check` (consult `--help` if you need)
4. Follow the steps to enable the `deviceOS@source` toolchain within Particle Workbench ([docs](https://docs.particle.io/support/particle-tools-faq/workbench/#working-with-a-custom-device-os-build))
5. Follow the guide to compile and flash your firmware project using Particle Workbench ([docs](https://docs.particle.io/tutorials/developer-tools/workbench/#local-build-and-flash))
6. Observe [passing CI workflow](https://app.circleci.com/pipelines/github/particle-iot/device-os/677/workflows/f19d2d33-1443-41d0-88d3-47b640ef1caa)


**Outcome**

Toolchain should be installed and validated, you should be able to compile and flash via Particle Workbench while using the `deviceOS@source` toolchain. 

### References

https://app.shortcut.com/particle/story/101084/e404x-define-new-platform-and-platform-id-if-needed
https://github.com/particle-iot/device-os/pull/2443
https://github.com/particle-iot/device-os/pull/2449

---

### Completeness
